### PR TITLE
Fix: Missing parameter properties info in constructors (fixes #143)

### LINKED ISSUE #143: Parameter properties information not being capture in ESTree parameter node
**What version of TypeScript are you using?**
2.0.10 and 2.1.5

**What version of `typescript-eslint-parser` are you using?**
1.0.2

**What code were you trying to parse?**

```ts
class Foo {
    constructor(public name : string) {}
}
```

```ts
class Foo {
    constructor(protected name : string) {}
}
```

```ts
class Foo {
    constructor(private name : string) {}
}
```

```ts
class Foo {
    constructor(readonly name : string) {}
}
```

```ts
class Foo {
    constructor(public readonly name : string) {}
}
```

```ts
class Foo {
    constructor(protected readonly name : string) {}
}
```

```ts
class Foo {
    constructor(private readonly name : string) {}
}
```

**What did you expect to happen?**
Parameter properties should be parsed correctly and modifiers should be part of the final ESTree parameter node.

**What happened?**
The modifiers property of the TS Node is being ignored by the AST Converter. The final ESTree contains the parameters, but without the accessibility modifiers and/or readonly modifiers.

Sources: [Parameter properties](http://www.typescriptlang.org/docs/handbook/classes.html#parameter-properties)

### PR DESCRIPTION
**Pull request by weirdpattern**
Includes modifications to ast-converter and corresponding tests

### CODE REVIEW

**Comment by weirdpattern** [comment]
Build is not being updated to complete, although travis completed successfully...

**Comment by soda0289** [comment]
@weirdpattern Maybe a webhook failed from traivs. No worries.

LGTM, Thanks!

**Comment by soda0289** [comment]
@JamesHenry Could you take a look at this one too. It would allow for us to implement the TSLint rule no-parameter-properties for the typescript plugin.

**Comment by JamesHenry** [comment]
@weirdpattern @soda0289 Sorry to be a pain, but could we do an initial PR which adds a test for a standard JS class with one or more parameters in its constructor?

I was surprised to see that we don't already have one in our ecmaFeatures/classes fixtures.

If we could get that merged in first, it would be a really useful reference within this PR to see a before and after. It would also make it much easier to identify and unintended issues.

**Comment by JamesHenry** [comment]
There may well be no change

**Comment by weirdpattern** [comment]
sure, no problem... I can work on this tonight unless @soda0289 wants to do it now, I leave the decision to you 😀

**Comment by soda0289** [comment]
I can write some tests, I don't mind

**Comment by weirdpattern** [comment]
@soda0289 go for it and thank you!

**Comment by JamesHenry** [comment]
Ready to be rebased against master! 😄

**Comment by soda0289** [comment]
@weirdpattern I think you did a merge instead of a rebase and force push

**Comment by weirdpattern** [comment]
@soda0289 I think you are right, I was sure I did a rebase, no idea what I did wrong ... what should I do now?

**Comment by soda0289** [comment]
@weirdpattern Perfrom the rebase again it should clean it up. When you perfrom the push it might warn you that the histroy has changed. Just ignore the message and do a git push --force

**Comment by weirdpattern** [comment]
@soda0289 ok I think I got it right this time... please review

**Comment by soda0289** [comment]
@weirdpattern Yup! LGTM, Thanks!

File: lib/ast-converter.js [hunk 1]
```diff
@@ -1187,10 +1187,27 @@ module.exports = function(ast, extra) {
...
                             }) : [];
-                            return convertedParam;
+
+                            if (param.modifiers) {
+                                return {
+                                    type: "TSParameterProperty",
+                                    range: [param.getStart(), param.end],
+                                    loc: getLoc(param, ast),
+                                    accessibility: getTSNodeAccessibility(param),
+                                    isReadonly: param.modifiers.filter(function(modifier) {
```
Comment (JamesHenry) [inline lib/ast-converter.js]: Using filter is a bit of an odd way to check this. More semantic would be:

```
isReadonly: param.modifiers.some(function(modifier) {
  return modifier.kind === SyntaxKind.ReadonlyKeyword
})
```

**Comment by weirdpattern** [comment]
@JamesHenry done

**Pull request merged**